vm.tf

```diff
@@ -0,0 +1,71 @@
+resource "azurerm_virtual_machine" "main" {
+  name                  = "${var.prefix}-vm"
+  location              = data.azurerm_resource_group.source_for_rg.location
+  resource_group_name   = data.azurerm_resource_group.source_for_rg.name
+  network_interface_ids = [azurerm_network_interface.main.id]
+  vm_size               = "Standard_DS1_v2"
+
+  # Uncomment this line to delete the OS disk automatically when deleting the VM
+  # delete_os_disk_on_termination = true
+
+  # Uncomment this line to delete the data disks automatically when deleting the VM
+  # delete_data_disks_on_termination = true
+
+  provisioner "local-exec" {
+    command = <<EOT
+    touch index.html && echo 'This is the default page for inginx' > index.html
+  EOT
+  }
+
+  provisioner "file" {
+    source      = "./index.html"
+    destination = "/home/testadmin/index.html"
+
+    connection {
+      type     = "ssh"
+      user     = "testadmin"
+      password = "Password1234!"
+      host     = azurerm_public_ip.public_ip.ip_address
+    }
+  }
+
+  provisioner "remote-exec" {
+    connection {
+      type     = "ssh"
+      user     = "testadmin"
+      password = "Password1234!"
+      host     = azurerm_public_ip.public_ip.ip_address
+    }
+
+    inline = [
+      "sudo apt-get update -y",
+      "sudo apt-get install -y nginx",
+      "sudo mv /home/testadmin/index.html /var/www/html/index.html"
+    ]
+  }
+
+  storage_image_reference {
+    publisher = "Canonical"
+    offer     = "0001-com-ubuntu-server-jammy"
+    sku       = "22_04-lts"
+    version   = "latest"
+  }
```

Comment:
Values `publisher, offer, sku, version` looks like reusable and will be better to move them in  `variables.tf` file

Reply:
fixed